hadar/lib/main.dart

```diff
@@ -18,12 +19,13 @@ void main() async {
   WidgetsFlutterBinding.ensureInitialized();
   await Firebase.initializeApp();
   const double marginSize = kIsWeb ? 0.0 : 0.0;
+  Widget logged_in_user = await CurrentUser.init_user();
   runApp(MaterialApp(
     debugShowCheckedModeBanner: false,
     home: Container(
         margin: const EdgeInsets.only(left: marginSize, right: marginSize),
         child: Scaffold(
-          body: LogInPage(),
+          body: logged_in_user != null ? logged_in_user : LogInPage(),
```

Comment:
Concisely: `logged_in_user ?? LogInPage()`

Reply:
I will change it , and close the pull request.
thank you!

Reply:
I wont close it , I have to add more the loading screen. for better user experience.
thank you anyway!

Reply:
There's no need to close anything. Simply make the change and push.